popkorn/src/androidMain/kotlin/ViewModel.kt

```diff
@@ -0,0 +1,14 @@
+package cc.popkorn.android.viewModel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewModelStoreOwner
+import cc.popkorn.popKorn
+
+class PopKornViewModelFactory : ViewModelProvider.Factory {
+    override fun <T : ViewModel> create(modelClass: Class<T>) = popKorn().inject(modelClass.kotlin)
+}
```

Comment:
If `PopKornViewModelFactory` should be internal, we can move it to a shared project.

Another question I have is if the new `ViewModel` APIs should belong to `:popkorn` or it should be moved to new module, for example`:popkorn-viewmodel`. I think this separated artifact is the right way because with multiplatform many people will use their own presentation layer without `ViewModel` API.

If we create a separate artifact for this, we should do the same for Compose and `ViewModel` support, `:popkorn-compose-viewmodel`